### PR TITLE
 Wait dynamically instead of wasting 120 seconds

### PR DESCRIPTION
**Pull request by MalloZup**
# What does this PR:

1) Introduce function which will wait dynamically https://github.com/kubic-project/automation/pull/529/files#diff-611e4142df421de6a9c053afa84ae6d8R13

Usage in codebase: 
2) remove sleep120
3) remove other sleeps

### CODE REVIEW

File: velum-bootstrap/spec/features/06-node-add.rb [hunk 1]
```diff
@@ -99,3 +94,17 @@
     puts "<<< Node add orchestration succeeded"
   end
 end
+
+def wait_until_node_appear(timeout)
+  puts ">>> Wait for new-node link to be enabled for max #{timeout} seconds"
+  Timeout.timeout(timeout) do
+    loop do
+      found_link = expect(page).to have_link(class: "assign-nodes-link")
+      break if found_link
```
Comment (dannysauer) [inline velum-bootstrap/spec/features/06-node-add.rb]: would this be ruby-er if written as `loop until  expect...`?  I always get nervous when I see `break`. :)

Comment (MalloZup) [reply velum-bootstrap/spec/features/06-node-add.rb]: thx nice ! i will fix it

File: velum-bootstrap/spec/features/06-node-add.rb [hunk 2]
```diff
@@ -99,3 +94,15 @@
     puts "<<< Node add orchestration succeeded"
   end
 end
+
+def wait_until_node_appear(timeout)
```
Comment (MaximilianMeister) [inline velum-bootstrap/spec/features/06-node-add.rb]: * i'd say `wait_until_node_appears`
* maybe move the function over to https://github.com/kubic-project/automation/blob/master/velum-bootstrap/spec/support/helpers.rb

File: velum-bootstrap/spec/support/helpers.rb [hunk 3]
```diff
@@ -6,6 +6,21 @@ def with_screenshot(name:)
...
+  # otherwise fail
+  def wait_until_true(timeout:)
+    puts ">>> Wait for new-node link to be enabled for max #{timeout} seconds"
+    Timeout.timeout(timeout) do
+      until yield do
+        sleep 1
+      end
+    end
+    rescue Timeout::Error
+      puts "FAIL: timoeut gone and function did not return true"
```
Comment (MalloZup) [inline velum-bootstrap/spec/support/helpers.rb]: i am open for better error msg :smile:

Comment (dannysauer) [reply velum-bootstrap/spec/support/helpers.rb]: "Error: timeout expired before link enabled"?

Comment (MalloZup) [reply velum-bootstrap/spec/support/helpers.rb]: Looks professional thx :)

File: velum-bootstrap/spec/features/06-node-add.rb [hunk 4]
```diff
@@ -30,19 +31,15 @@
...
-    sleep 120
-    puts "<<< Waiting 120 seconds as a workaround"
-
     # wait for new link to appear
-    puts ">>> Wait for new-node link to be enabled"
+    max_timeout = 240
     with_screenshot(name: :new_node_link_enabled) do
-      expect(page).to have_link(class: "assign-nodes-link", wait: 120)
+      wait_until_true(timeout: max_timeout) do
+	      expect(page).to have_link(class: "assign-nodes-link"
```
Comment (MaximilianMeister) [inline velum-bootstrap/spec/features/06-node-add.rb]: 2 whitespaces indentation should be enough ;)

File: velum-bootstrap/spec/features/04-upgrade-admin.rb [hunk 5]
```diff
@@ -73,3 +59,13 @@
...
   end
 end
+
+def wait_velum_recover
+  with_status_ok do
+    visit "/"
+  end
+  true
+rescue Exception => e
+    puts "... still waiting for velum to recover"
```
Comment (MaximilianMeister) [inline velum-bootstrap/spec/features/04-upgrade-admin.rb]: same here with indentation

File: velum-bootstrap/spec/features/06-node-add.rb (hunk 4 above)
Comment (MalloZup) [reply velum-bootstrap/spec/features/06-node-add.rb]: :japanese_castle:

Comment (MalloZup) [reply velum-bootstrap/spec/features/06-node-add.rb]: @MaximilianMeister  use emoticons for smile please , see the CoC ( code of conduct :smile: )

**Comment by MalloZup** [comment]
https://github.com/kubic-project/automation/issues/534

I have the feeling i will never get this CI green here :hankey: :crying_cat_face: :sob:

File: velum-bootstrap/spec/support/helpers.rb [hunk 6]
```diff
@@ -6,6 +6,21 @@ def with_screenshot(name:)
...
+  def wait_until_true(timeout:)
+    puts ">>> Wait for new-node link to be enabled for max #{timeout} seconds"
+    Timeout.timeout(timeout) do
+      until yield do
+        sleep 1
+      end
+    end
+    rescue Timeout::Error
+      puts "Error: timeout reached before function was successfull"
+      false
```
Comment (MaximilianMeister) [inline velum-bootstrap/spec/support/helpers.rb]: Line `20-22` need to be indented to the left

File: velum-bootstrap/spec/features/06-node-add.rb (hunk 4 above)
Comment (MaximilianMeister) [reply velum-bootstrap/spec/features/06-node-add.rb]: i can still see wrong indentation here, 6 whitespaces instead of 2

Comment (MalloZup) [reply velum-bootstrap/spec/features/06-node-add.rb]: i reverted the commit with the fix and other things

**Comment by MalloZup** [comment]
@MaximilianMeister i am just closing because is not worth the effort and i other things todo but i found out some ci issues which is was ok . 


Thank you for your time :love_hotel:

**Pull request closed without merging**